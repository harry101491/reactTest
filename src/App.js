import { useState, useEffect } from "react";

function App() {
  const [flag, setFlag] = useState(false);
  const [label, setLabel] = useState('');
  
  const onButtonClick = () => {
    setFlag(!flag);
  }

  useEffect(() => {
    setLabel(flag ? 'Flag On' : 'Flag Off');
  }, [label, flag]);

  return (
    <div className="App">
      <p>{ label }</p>
      <button onClick={onButtonClick}>Change Flag</button>
    </div>
  );
}

export default App;
