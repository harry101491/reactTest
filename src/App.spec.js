import { render } from '@testing-library/react';
import App from './App';

describe('App Component', () => {
  
  it('should match snapshot', () => {
    const element = render(<App />).baseElement;
    expect(element).toMatchSnapshot();
  });
});
