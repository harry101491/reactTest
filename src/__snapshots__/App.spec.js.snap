// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snapshot 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <p>
        Flag Off
      </p>
      <button>
        Change Flag
      </button>
    </div>
  </div>
</body>
`;
